res/ari/resource_channels.c

```diff
@@ -1665,6 +1674,11 @@ static void ari_channels_handle_snoop_channel(
 		return;
 	}
 
+	if (!ast_strlen_zero(args_pre_dial_go_sub)) {
```

Comment:
This thread does not have ownership or control over the returned snoop channel. It is unsafe to execute here.